mlflow/metrics/metric_definitions.py

```diff
@@ -371,3 +361,19 @@ def _f1_score_eval_fn(
             sample_weight=sample_weight,
         )
         return MetricValue(aggregate_results={"f1_score": f1})
+
+
+def _precision_at_k_eval_fn(predictions, targets, k, metrics, sample_weight=None):
```

Comment:
Is it possible to pass `k` as a scalar value? Can we validate k is int and `k>0`?

Reply:
had to rebase to include sunish's latest PR, but it works now!